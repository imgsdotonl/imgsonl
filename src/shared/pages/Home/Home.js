/* @flow */

import React from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';

import PageTemplate from '../../theme/Imgsonl';
import { Grid, Row, Hero, Footer } from '../../components/index';
import { safeConfigGet } from '../../core/utils/config';

type Props = {
  loaded: Boolean,
  pages: Object,
  entities: Object,
  dispatch: Function
};

const Home = (props: Props) => {
  return (
      <PageTemplate
        helmetMeta={ <Helmet title="Home" /> }
        hero={ <Hero /> }
        footer={ <Footer /> }
      >
      <Grid fluid>
        <Row style={ { padding: '25px' } }>
          Placeholder
        </Row>
      </Grid>
      </PageTemplate>
  );
};

const mapStateToProps = (state) => {

};

export default connect()(Home);
