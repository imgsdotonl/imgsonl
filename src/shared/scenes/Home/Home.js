/* @flow */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import AppBar from 'material-ui/AppBar';
import Helmet from 'react-helmet';

import PageTemplate from '../../theme/Imgsonl';
import { Grid, Row, Col, Footer, FileUpload, UrlForm, FileViewer } from '../../components/index';
import { uploadFile, uploadUrl } from '../../state/modules/file';
import { goHome } from '../../state/modules/ui';
import Progress from '../../components/Progress';

type Props = {
  loaded: Boolean,
  handleUpload: Function,
  uploadFile: Function,
  uploadUrl: Function,
  goHome: Function,
  files: Object,
  dispatch: Function
};

class Home extends Component {
  constructor(props) {
    super();
    (this: any).handleSubmit = this.handleSubmit.bind(this);
    (this: any).handleUpload = this.handleUpload.bind(this);
    (this: any).navigateHome = this.navigateHome.bind(this);
  }

  props: Props;

  handleUpload(payload) {
    this.props.uploadFile(payload);
  }
  handleSubmit(values) {
    this.props.uploadUrl(values);
  }
  navigateHome() {
    this.props.goHome();
  }
  render() {
    return (
      <PageTemplate
        header={ <AppBar title="imgs.onl" titleStyle={ { cursor: 'pointer' } } showMenuIconButton={ false } /> }
        helmetMeta={ <Helmet title="Home" /> }
        footer={ <Footer /> }
      >
      <Grid>
        <Row>
            <Col xs={ 12 }>
              <Row xsCenter>
                <Col xs={ 6 }>
                  <FileUpload handleUpload={ this.handleUpload } />
                  <UrlForm onSubmit={ this.handleSubmit } />
                </Col>
              </Row>
            </Col>
        <Col sm={ 12 } md={ 6 }>
          {
            !this.props.files.file.filepath
            ? null
            : <FileViewer filepath={ this.props.files.file.filepath } />
          }
        </Col>
        </Row>
      </Grid>
      { this.props.progress.active ? <Progress message={ this.props.progress.message } /> : null }
      </PageTemplate>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    ui: state.ui,
    files: state.files,
    progress: state.progress,
  };
};

export default connect(mapStateToProps, { uploadFile, uploadUrl, goHome })(Home);
