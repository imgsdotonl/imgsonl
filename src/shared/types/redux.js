/* @flow */

export type ApiAction<T> = {
  key: string,
  payload: T,
};

export type Store = {
  dispatch(action: ActionType): ActionType
}
export type ActionType =
  | 'LOGIN_REQUEST'
  | 'LOGIN_SUCCESS'
  | 'LOGIN_FAILURE'
  | 'LOGOUT_USER'
  | 'LOGOUT_USER_FAIL'
  | 'CHECK_AUTH_REQUEST'
  | 'CHECK_AUTH_SUCCESS'
  | 'CHECK_AUTH_FAILURE'
  | 'SIGNUP_USER_REQUEST'
  | 'SIGNUP_USER_SUCCESS'
  | 'SIGNUP_USER_FAILURE'
  | 'FORGOT_PASSWORD_REQUEST'
  | 'FORGOT_PASSWORD_SUCCESS'
  | 'FORGOT_PASSWORD_FAILURE'
  | 'RESET_PASSWORD_REQUEST'
  | 'RESET_PASSWORD_SUCCESS'
  | 'RESET_PASSWORD_FAILURE'
  | 'LOAD_POSTS_REQUEST'
  | 'LOAD_POSTS_SUCCESS'
  | 'LOAD_POSTS_FAILURE'
  | 'GET_POST_REQUEST'
  | 'GET_POST_SUCCESS'
  | 'GET_POST_FAILURE'
  | 'UPDATE_POST_REQUEST'
  | 'UPDATE_POST_SUCCESS'
  | 'UPDATE_POST_FAILURE'
  | 'SELECT_POST'
  | 'SELECT_POST_SUCCESS'
  | 'SELECT_POST_FAILURE'
  | 'CREATE_POST_REQUEST'
  | 'CREATE_POST_SUCCESS'
  | 'CREATE_POST_FAILURE'
  | 'DELETE_POST_FAILURE'
  | 'DELETE_POST_REQUEST'
  | 'DELETE_POST_SUCCESS';
