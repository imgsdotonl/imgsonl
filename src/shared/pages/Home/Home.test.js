
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Home } from './Home';

describe('<Home />', () => {
  const store = createStore(() => ({}));
  it('renders <Home /> container', () => {
    const wrapper = shallow(<Provider store={ store }><Home /></Provider>);
    expect(wrapper.find('div').length).toBe(0);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(<Provider store={ store }><Home /></Provider>)).toMatchSnapshot();
  });
});
