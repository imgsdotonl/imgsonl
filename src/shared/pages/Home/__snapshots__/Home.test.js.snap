exports[`<Home /> renders snapshot 1`] = `undefined`;
